rpc/src/fetch.rs

```diff
@@ -28,3 +33,42 @@ pub async fn proxy_query_to_history_subnet(
         )),
     }
 }
+
+pub async fn find_header_by_hash(
+    network: &mpsc::UnboundedSender<HistoryJsonRpcRequest>,
+    block_hash: H256,
+) -> Result<Header, RpcServeError> {
+    // Request the block header from the history subnet.
+    let content_key: HistoryContentKey = HistoryContentKey::BlockHeaderWithProof(block_hash.into());
+    let endpoint = HistoryEndpoint::RecursiveFindContent(content_key);
+    let mut result = proxy_query_to_history_subnet(network, endpoint).await?;
+    let content = match result["content"].take() {
+        serde_json::Value::String(s) => s,
+        wrong_type => {
+            let message =
+                format!("Invalid internal representation of block header; json: {wrong_type:?}");
+            return Err(RpcServeError::Message(message));
+        }
+    };
+    if content == CONTENT_ABSENT {
+        return Err(RpcServeError::Message("Block not found".into()));
+    };
+    let content: Vec<u8> =
+        hex_decode(&content).expect("decoding the trin hex-encoded data failed, odd");
+    let header = match HistoryContentValue::decode(&content) {
+        Ok(header) => header,
+        Err(err) => {
+            let message =
+                format!("Invalid internal representation of block header; could not decode: {err}");
+            return Err(RpcServeError::Message(message));
+        }
+    };
+
+    match header {
+        HistoryContentValue::BlockHeaderWithProof(h) => Ok(h.header),
+        wrong_val => Err(RpcServeError::Message(format!(
+            "Found something other than block header at the key; got {:?}",
```

Comment:
I'm not sure I follow this error message. Looks like `at` is a typo for `as`. But also, we're dealing with a value here not the key.

Reply:
I mean, it shouldn't ever happen but... what it means is that we asked HistoryEndpoint to give us the content value for a content key that specifies a header, but then the endpoint returned a value that is not a header.

So that's what the error message is trying to say. Given that the content key is definitely a header key, something went very wrong with the HistoryEndpoint, and it gave us some other type of content.

Does this error message convey that any more clearly?
"Internal trin error: got back a non-header from a key that must only point to headers; got ..."